extensions/guacamole-auth-quickconnect/src/main/java/org/apache/guacamole/auth/quickconnect/QuickConnection.java

```diff
@@ -0,0 +1,181 @@
+import org.apache.guacamole.net.GuacamoleSocket;
+import org.apache.guacamole.net.GuacamoleTunnel;
+import org.apache.guacamole.net.InetGuacamoleSocket;
+import org.apache.guacamole.net.SSLGuacamoleSocket;
+import org.apache.guacamole.net.SimpleGuacamoleTunnel;
+import org.apache.guacamole.net.auth.AbstractConnection;
+import org.apache.guacamole.net.auth.Connection;
+import org.apache.guacamole.net.auth.ConnectionRecord;
+import org.apache.guacamole.net.auth.GuacamoleProxyConfiguration;
+import org.apache.guacamole.protocol.ConfiguredGuacamoleSocket;
+import org.apache.guacamole.protocol.GuacamoleClientInformation;
+import org.apache.guacamole.protocol.GuacamoleConfiguration;
+
+/**
+ * A type of Connection specific to this authentication extension.
+ */
+public class QuickConnection extends AbstractConnection {
+
+    /**
+     * Backing configuration, containing all sensitive information.
+     */
+    private GuacamoleConfiguration config;
+
+    /**
+     * Number of active connections.
+     */
+    private int activeConnections;
+
+    /**
+     * Empty connection constructor.
+     */
+    public QuickConnection() {
+
+    }
+
+    /**
+     * Constructor that takes a name, identifier, and GuacamoleConfiguration
+     * and builds a QuickConnection from it.
+     *
+     * @param name
+     *     The name of the connection.
+     *
+     * @param identifier
+     *     The unique identifier of this connection within this
+     *     authentication module.
+     *
+     * @param config
+     *     The GuacamoleConfiguration object to store in this
+     *     QuickConnection.
+     */
+    public QuickConnection(String name, String identifier,
+            GuacamoleConfiguration config) {
+
+        setName(name);
+
+        setIdentifier(identifier);
+
+        setConfiguration(config);
+        this.config = config;
+
+        this.activeConnections = 0;
+
+    }
+
+    /**
+     * Constructs a QuickConnection from a generic Connection
+     * object, copying over the relevant data and initializing
+     * the rest.
+     *
+     * @param object
+     *     The generic Connection object to be copied.
+     */
+    public QuickConnection(Connection object) {
+
+        setName(object.getName());
+        setIdentifier(object.getIdentifier());
+        setParentIdentifier(object.getParentIdentifier());
+        setConfiguration(object.getConfiguration());
+        this.config = object.getConfiguration();
+        this.activeConnections = 0;
+
+    }
+
+    @Override
+    public int getActiveConnections() {
+        return activeConnections;
+    }
+
+    @Override
+    public Map<String, String> getAttributes() {
+        return Collections.<String, String>emptyMap();
+    }
+
+    @Override
+    public void setAttributes(Map<String, String> attributes) {
+        // Do nothing - there are no attributes
+    }
+
+    @Override
+    public GuacamoleTunnel connect(GuacamoleClientInformation info)
+            throws GuacamoleException {
+
```

Comment:
This is not a comment about the quick connect extension so much as a comment about whether there is a generalization here that should be abstracted. There's a body of about 40 lines of code here that has nothing to do with quick connect. Isn't this behavior (using the environment to get the proxy config and figure out what's needed to create an appropriate Guacamole socket) something that should be either pulled up into the abstract base type or otherwise moved into a reusable component?

Reply:
Yes, you're probably right, and it's probably already implemented in some other connection.  Let me see if there's some other class that I should be `extend`ing or `implement`ing that already has this rather than having it in here.

Reply:
So, I think when I originally wrote this I copied the `org.apache.guacamole.net.auth.simple.SimpleConnection` class over to this QuickConnection class, instead of just extending it.  I've refactored it to just extend the SimpleConnection object and override where I need to instead of duplicaitng the code.  Thanks for pointing that out!